org.jacoco.core.test/src/org/jacoco/core/test/filter/targets/Finally.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.filter.targets;
+
+import static org.jacoco.core.test.validation.targets.Stubs.nop;
+
+/**
+ * This test target is a finally block.
+ */
+public class Finally {
```

Comment:
@marchof About control structures such as `if`: there is already test of conditional jump that is generated for `nop(i == 0)`, so I'm hesitating with addition of test for `if` statement - in this case why not add tests for other statements? and what about their relative order and so on? i.e. how to not go mad trying to generate all possible variants that are infinite in general.

Nested `finally` in his turn shows difference of slots for different duplicates of same `finally` block, which is not shown in existing tests here, so added it.